lib/trino-parquet/src/main/java/io/trino/parquet/reader/ParquetReader.java

```diff
@@ -363,6 +366,7 @@ private ColumnChunk readPrimitive(PrimitiveField field)
             OffsetIndex offsetIndex = getFilteredOffsetIndex(currentRowGroup, currentBlockMetadata.getRowCount(), metadata.getPath());
             List<Slice> slices = allocateBlock(fieldId);
             columnReader.setPageReader(createPageReader(slices, metadata, columnDescriptor, offsetIndex), currentGroupRowRanges);
+            codecsUsed.add(metadata.getCodec().name());
```

Comment:
Let's do this in the constructor instead where we loop over metadata of row groups